common/djangoapps/enrollment/tests/test_data.py

```diff
@@ -256,3 +257,58 @@ def test_update_for_non_existent_user(self):
     def test_update_for_non_existent_course(self):
         enrollment = data.update_course_enrollment(self.user.username, "some/fake/course", is_active=False)
         self.assertIsNone(enrollment)
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Test only valid in lms')
+class RosterDataTest(ModuleStoreTestCase):
+    """
+    Test course roster data aggregation api.
+    """
+    STUDENT_USERNAME = 'aBob'
+    STUDENT_NAME = 'Bob Student'
+    STUDENT_EMAIL = 'bob@example.com'
+    STUDENT_PASSWORD = 'edx'
+    STAFF_USERNAME = 'zJoe'
+    STAFF_NAME = 'Joe Staff'
+    STAFF_EMAIL = 'joe@example.com'
+    STAFF_PASSWORD = 'edx'
+
+    def setUp(self):
+        """
+        Create a course, and enroll a student user and staff user.
+        """
+        super(RosterDataTest, self).setUp()
+        self.course = CourseFactory.create()
+        self.course_key = self.course.id
+
+        self.student = UserFactory.build(
+            username=self.STUDENT_USERNAME,
+            email=self.STUDENT_EMAIL,
+            password=self.STUDENT_PASSWORD,
+        )
+        self.student.save()
+        UserProfileFactory.create(user=self.student, name=self.STUDENT_NAME)
+
+        self.staff = UserFactory.build(
+            username=self.STAFF_USERNAME,
+            email=self.STAFF_EMAIL,
+            password=self.STAFF_PASSWORD,
+        )
+        self.staff.save()
+        UserProfileFactory.create(user=self.staff, name=self.STAFF_NAME)
+
+        CourseEnrollment.enroll(self.student, self.course_key, mode="honor")
+        CourseEnrollment.enroll(self.staff, self.course_key, mode="honor")
+        CourseStaffRole(self.course_key).add_users(self.staff)
+
+    def test_roster(self):
+
+        roster = data.get_roster(self.course_key.to_deprecated_string())
```

Comment:
New code should not call `to_deprecated_string`;
isn't the new syntax just `unicode(self.course_key)`?